### PR TITLE
K8SPS-176: remove redundant volume mounts for delete backup job

### PR DESCRIPTION
**Pull request by pooknull**
[K8SPS-176](https://jira.percona.com/browse/K8SPS-176) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

https://jira.percona.com/browse/K8SPS-176

The `delete-backup` finalizer fails if a cluster was deleted before, because of the missing `cluster1-ssl` secret. This secret has owner references to the cluster object and gets deleted during cluster deletion.

When the cluster is missing, the `delete-backup` finalizer creates a job that executes `xbcloud` with arguments for deletion. So, we don't need this `cluster1-ssl` secret and any other volume mounts which previously contained this job.

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-226/5a8e007/init-deploy.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-226/5a8e007/demand-backup.log) |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-226/5a8e007/auto-config.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-226/5a8e007/gr-init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-226/5a8e007/monitoring.log) |
| semi-sync | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-226/5a8e007/semi-sync.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-226/5a8e007/service-per-pod.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-226/5a8e007/config.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-226/5a8e007/sidecars.log) |
| gr-demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-226/5a8e007/gr-demand-backup.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-226/5a8e007/limits.log) |
| version-service | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-226/5a8e007/version-service.log) |
| one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-226/5a8e007/one-pod.log) |
| tls-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-226/5a8e007/tls-cert-manager.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-226/5a8e007/scaling.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-226/5a8e007/users.log) |
| We run 16 out of 16|

commit: https://github.com/percona/percona-server-mysql-operator/pull/226/commits/5a8e0070e1a990449a1535cc097d8c66c5ffac90
image: `perconalab/percona-server-mysql-operator:PR-226-5a8e007`

File: pkg/xtrabackup/xtrabackup.go [hunk 1]
```diff
@@ -273,14 +273,6 @@ func deleteContainer(image string, conf *BackupConfig, storage *apiv1alpha1.Back
 				Name:      apiv1alpha1.BinVolumeName,
 				MountPath: apiv1alpha1.BinVolumePath,
 			},
-			{
```
Comment (hors) [inline pkg/xtrabackup/xtrabackup.go:276]: @egegunes what do you think about it? We do not use it now, so we can remove it. Then we will see

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-226/5a8e007/auto-config.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-226/5a8e007/config.log) |
| one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-226/5a8e007/one-pod.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-226/5a8e007/init-deploy.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-226/5a8e007/gr-init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-226/5a8e007/monitoring.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-226/5a8e007/demand-backup.log) |
| semi-sync | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-226/5a8e007/semi-sync.log) |
| gr-demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-226/5a8e007/gr-demand-backup.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-226/5a8e007/service-per-pod.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-226/5a8e007/scaling.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-226/5a8e007/sidecars.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-226/5a8e007/users.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-226/5a8e007/limits.log) |
| version-service | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-226/5a8e007/version-service.log) |
| tls-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-226/5a8e007/tls-cert-manager.log) |
| We run 16 out of 16|

commit: https://github.com/percona/percona-server-mysql-operator/pull/226/commits/5a8e0070e1a990449a1535cc097d8c66c5ffac90
image: `perconalab/percona-server-mysql-operator:PR-226-5a8e007`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-226/4ef334d/auto-config.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-226/4ef334d/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-226/4ef334d/init-deploy.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-226/4ef334d/gr-init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-226/4ef334d/monitoring.log) |
| semi-sync | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-226/4ef334d/semi-sync.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-226/4ef334d/service-per-pod.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-226/4ef334d/config.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-226/4ef334d/sidecars.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-226/4ef334d/limits.log) |
| gr-demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-226/4ef334d/gr-demand-backup.log) |
| version-service | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-226/4ef334d/version-service.log) |
| one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-226/4ef334d/one-pod.log) |
| tls-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-226/4ef334d/tls-cert-manager.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-226/4ef334d/scaling.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-226/4ef334d/users.log) |
| We run 16 out of 16|

commit: https://github.com/percona/percona-server-mysql-operator/pull/226/commits/4ef334dba7eca0431285a293feeee6d826f0bb7c
image: `perconalab/percona-server-mysql-operator:PR-226-4ef334d`

**Pull request merged**